lib/css/CssParser.js

```diff
@@ -368,6 +371,74 @@ class CssParser extends Parser {
 			pos = walkCssTokens.eatWhiteLine(input, pos);
 			return pos;
 		};
+		const parseImports = (input, start, pos) => {
+			let url = "";
+			pos = walkCssTokens.eatWhitespaceAndComments(input, pos);
+			[pos, url] = eatText(input, pos, eatUntil(")"));
+			pos++;
+			const data = { url, range: [0, 0], imports: {} };
+			pos = walkCssTokens.eatWhitespaceAndComments(input, pos);
+			const cc = input.charCodeAt(pos);
+			if (cc !== CC_LEFT_CURLY) {
+				this._emitWarning(
+					state,
+					`Unexpected '${input[pos]}' at ${pos} during parsing of ':export' (expected '{')`,
+					locConverter,
+					pos,
+					pos
+				);
+				return pos;
+			}
+			pos++;
+			pos = walkCssTokens.eatWhitespaceAndComments(input, pos);
+			for (;;) {
+				if (input.charCodeAt(pos) === CC_RIGHT_CURLY) break;
+				pos = walkCssTokens.eatWhitespaceAndComments(input, pos);
+				if (pos === input.length) return pos;
+				let start = pos;
+				let name;
+				[pos, name] = eatText(input, pos, eatExportName);
+				if (pos === input.length) return pos;
+				if (input.charCodeAt(pos) !== CC_COLON) {
+					this._emitWarning(
+						state,
+						`Unexpected '${input[pos]}' at ${pos} during parsing of export name in ':import' (expected ':')`,
+						locConverter,
+						start,
+						pos
+					);
+					return pos;
+				}
+				pos++;
+				if (pos === input.length) return pos;
+				pos = walkCssTokens.eatWhitespaceAndComments(input, pos);
+				if (pos === input.length) return pos;
+				let value;
+				[pos, value] = eatText(input, pos, eatExportValue);
+				if (pos === input.length) return pos;
+				const cc = input.charCodeAt(pos);
+				if (cc === CC_SEMICOLON) {
+					pos++;
+					if (pos === input.length) return pos;
+					pos = walkCssTokens.eatWhitespaceAndComments(input, pos);
+					if (pos === input.length) return pos;
+				} else if (cc !== CC_RIGHT_CURLY) {
+					this._emitWarning(
+						state,
+						`Unexpected '${input[pos]}' at ${pos} during parsing of export value in ':import' (expected ';' or '}')`,
+						locConverter,
+						start,
+						pos
+					);
+					return pos;
+				}
+				data.imports[name] = { value, replaceRanges: [] };
+			}
+			pos++;
+			data.range = [start, pos];
+			icssImportData.push(data);
+			return pos;
+		};
```

Comment:
Avoid such things, you can use set `let inIcssImportSelector` when you found `:import`, and collect all things, you can use more `is*` helpers, avoid any changes in tokenizer logic

Reply:
Thanks for your suggestion, these code is inspired by `parseExports`, maybe it needs to be changed as well.